1.0/docs/devguide/templates.md

```diff
@@ -166,6 +171,83 @@ Changing a `manager.type` field should now cause the list to be re-sorted:
 
     this.set('employees.0.manager.type', 'engineer');
 
+#### Dynamic sort and filter changes
+
+The `observe` property lets you specify item sub-properties to 
+observe for filtering and sorting purposes. However, sometimes you want to
+dynamically change the sort or filter based on another unrelated value. In 
+this case, you can use a computed binding to _return_ a dynamic filter or 
+sort function when one or more dependent properties changes.
+
+    <dom-module id="employee-search">
+
+      <template>{% raw %}
+        <input value="{{searchString::change}}">
+        <template is="dom-repeat" items="{{employees}}" as="employee"
+            filter="{{computeFilter(searchString)}}">
+            <div>{{employee.lastname}}, {{employee.firstname}}</div>
+        </template>{% endraw %}
+      </template>
+
+      <script>
+        Polymer({
+          is: "employee-search",
+          computeFilter: function(string) {
+            if (!string) {
+              // set filter to null to disable filtering
+              return null;
+            } else {
+              // return a filter function for the current search string
+              string = string.toLowerCase();
+              return function(employee) {
+                var first = employee.firstname.toLowerCase();
+                var last = employee.lastname.toLowerCase();
+                return first.includes(string) ||
```

Comment:
FYI, `String.prototype.includes` is ES6, and won't work in IE afaik.  Might be better to use ES5 here (`indexOf` or `match` or something like that)...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes